src/models/loan/paymentCalculations.js

```diff
@@ -0,0 +1,46 @@
+function daysInYear(year) {
+  return 365
+}
+
+function daysInMonth(month) {
+  return 30.417
+}
+
+function effectiveMonthlyInterestRate(yearlyInterestRate, month, year) {
+  return ((1 + yearlyInterestRate / daysInYear(year)) ** daysInMonth(month)) - 1
+}
+
+const calculateRepaymentTerm = ({debt:{amountOwed, interestRate: {rate}}, paymentPlan: {monthlyPayment}}) => {
+  let repaymentTerm
+  const interestRate = effectiveMonthlyInterestRate(rate)
+  if (amountOwed == 0 ) {
+    repaymentTerm = 0
+  } else if (rate) {
+    if (monthlyPayment <= amountOwed * interestRate && amountOwed > 0) {
+      repaymentTerm = Infinity
+    } else {
+      repaymentTerm = (Math.log(monthlyPayment)
+                        - Math.log(monthlyPayment - amountOwed * effectiveMonthlyInterestRate(rate))
+                      ) / Math.log(1 + effectiveMonthlyInterestRate(rate))
```

Comment:
why not reuse the `interestRate` variable above?

Reply:
actually why not move that declaration down here, since it also doesn't make much sense to call `effectiveMonthlyInterestRate` on something that may not exist (given your `else if` above)